components/Footer/Footer.tsx

```diff
@@ -0,0 +1,74 @@
+import clsx from "clsx";
+import Image from "next/image";
+import Link from "next/link";
+import { HTMLAttributes } from "react";
+
+import { getLinkPath } from "@/lib/menus";
+import { usePaths } from "@/lib/paths";
+import { useFooterMenuQuery } from "@/saleor/api";
+
+import { Box } from "../Box";
+import { useRegions } from "../RegionsProvider";
+import styles from "./Footer.module.css";
+
+export type FooterProps = HTMLAttributes<HTMLElement>;
+
+export function Footer({ className, ...rest }: FooterProps) {
+  const paths = usePaths();
+  const { query, currentChannel, currentLocale } = useRegions();
+
+  const { data, error } = useFooterMenuQuery({ variables: { ...query } });
+
+  if (error) {
+    console.error("Footer component error", error.message);
+    return null;
+  }
```

Comment:
I'd just print the error to console but not return early, since there are already guards against `data` not being defined

Reply:
Removed line with null return.